contracts/clients/src/SafeSingletonFactory.ts

```diff
@@ -63,11 +64,24 @@ export default class SafeSingletonFactory {
     },
   };
 
-  constructor(
+  provider: ethers.providers.Provider;
+
+  // eslint-disable-next-line no-use-before-define
+  viewer: SafeSingletonFactoryViewer;
+
+  private constructor(
     public signer: ethers.Signer,
```

Comment:
Why is a signer now required for this? Someone may want to use this in a read-only mode for safety, and it looks like `SafeSingletonFactoryViewer` can work with just a provider.

Reply:
I think the diff is just a bit confusing here - it was requiring a signer previously.

This PR actually introduces `SafeSingletonFactoryViewer` for this purpose, which allows you to create it with `signerOrProvider`.